adobe/spectrum-web-components
# fix(tabs): scroll exceeding tabs limit

## Description

This PR aims to address said problem by limitting the delta a consumer can pass to the `scrollTabs` public method, used in `sp-tabs-overflow`.

## Related issue(s)

- https://github.com/adobe/spectrum-web-components/issues/4700

## Motivation and context

Currently on iOS devices the user can still act upon the "left/right" chevron icons to scroll through the tabs even when there are no more tabs to scroll to. Also, an empty space is visible when scrolling to the right/left most tab.

## How has this been tested?

    1. Go to the tabs overflow storybook using an iOS device
    2. Click on the "right" icon to scroll to the right until you reach the last tab item
    3. Observe that the scroll stops right at the last item, without any empty spaces and the icon disappears
    1. Repeat _test case 1_
    2. Click on the "left" icon to scroll to the left until you reach the first item
    3. Observe that the scroll stops right at the first item, without any empty spaces and the icon disappears
    1. Go to the tabs overflow storybook using an iOS device
    2. Set the direction to 'rtl'
    3. Click on the "left" icon to scroll to the left until you reach the last tab item
    4. Observe that the scroll stops right at the last item, without any empty spaces and the icon disappears
    1. Repeat _test case 3_
    2. Click on the "right" icon to scroll to the right until you reach the first item
    3. Observe that the scroll stops right at the first item, without any empty spaces and the icon disappears


## Screenshots (if appropriate)


https://github.com/user-attachments/assets/eb0cd502-b928-4949-bd48-acec6209a966


## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
